komfydb/execution/predicate.cc

```diff
@@ -0,0 +1,55 @@
+#include "komfydb/execution/predicate.h"
+
+namespace {
+
+using namespace komfydb::common;
+using namespace komfydb::storage;
+
+}  // namespace
+
+Predicate::Predicate(Op op, int field, std::unique_ptr<Field> const_field)
+    : op(op),
+      field1(field),
+      const_field(std::move(const_field)),
+      type(Type::COL_CONST) {}
+
+Predicate::Predicate(Op op, int field1, int field2)
+    : op(op),
+      field1(field1),
+      field2(field2),
+      const_field(nullptr),
+      type(Type::COL_COL) {}
+
+Op Predicate::GetOp() {
+  return op;
+}
+
+Field* Predicate::GetConstField() {
+  return const_field.get();
+}
+
+int Predicate::GetField1() {
+  return field1;
+}
+
+int Predicate::GetField2() {
+  return field2;
+}
+
+Predicate::Type Predicate::GetType() {
+  return type;
+}
+
+absl::StatusOr<bool> Predicate::Evaluate(const Record& record) {
```

Comment:
I think no need to return `absl::StatusOr<bool>`, I think it's safe to assume GetField shouldn't fail here.